### DISCUSSION TITLE
Does the Lennox API report energy use by the HVAC system?

### DISCUSSION THREAD

**Discussion by dcj** [PeteRager/lennoxs30#54] [Ideas]
If so, it would be nice to tie that into the home-assistant energy reporting, but I'm guessing that the API doesn't report this....

**Comment by PeteRager** [comment]
That’s a great question.   As far as I can tell it does not provide the information and I see nothing in the Lennox docs that indicate otherwise.  You’d think they’d have a cloud dashboard showing this info?

 Mine has two circuits - a 50 amp for the compressor and a 30 amp for the air handler and hence would need a pair of current sensors.

**Comment by PeteRager** [comment]
Interestingly we may have found a way to do this.  Follow along with this issue

https://github.com/PeteRager/lennoxs30/issues/70

**Comment by highincolorado** [comment]
Adding to this thread, our unit is a heat pump with backup propane furnace. I love that I can now get historical data about run time but would love to know which heat source is running since it can use either or both. Getting power and (dreaming here) fuel usage would allow me to fine tune setting based upon fuel cost. 
Thanks for your efforts!

**Reply by PeteRager** [reply]
> Adding to this thread, our unit is a heat pump with backup propane furnace. I love that I can now get historical data about run time but would love to know which heat source is running since it can use either or both. Getting power and (dreaming here) fuel usage would allow me to fine tune setting based upon fuel cost.
> Thanks for your efforts!

That information should be available in the system status messages that we get from Lennox. Just a matter of figuring out what it is.   If you could capture debug logs when the heat pump turns on and when the backup propane furnace comes on - that should give us the info needed to add it it.  It may also be be helpful to get the initial config messages that come in (100kb) as they would have the information on your system config.

So I guess, put the integration into debug mode, restart HA, set the heat so the heat pump comes on, then make the backup furnace come on.

**Reply by highincolorado** [reply]
I’ll work on it tomorrow.

**Reply by highincolorado** [reply]
Ok I have a lot of log data, where can I send the file and notes. Not sure I was ever able to stop the heat pump from running no matter high or low balance points but the gas furnace certainly ran. I believe "aux": true is the message that tells that whatever the aux heat system is is running. I scrubbed as much personal data as I could find from the log file but don't feel comfortable posting here publicly since people more informed than me might be able to grab something I missed.

Sad the lennox data is not logged in it's own files..

**Reply by PeteRager** [reply]
> Ok I have a lot of log data, where can I send the file and notes. Not sure I was ever able to stop the heat pump from running no matter high or low balance points but the gas furnace certainly ran. I believe "aux": true is the message that tells that whatever the aux heat system is is running. I scrubbed as much personal data as I could find from the log file but don't feel comfortable posting here publicly since people more informed than me might be able to grab something I missed.
> 
> Sad the lennox data is not logged in it's own files..

Great news.  Yeah, the Aux flag is probably it.  Email log to me.  Pete dot sage at iCloud dot com
I have thought of putting message logs into own files, I’ll add that to the list.  Here’s an issue for Auxiliary heat, let’s use that to collaborate on the solution. 

https://github.com/PeteRager/lennoxs30/issues/85
